library/include/rocwmma/internal/dpp.hpp

```diff
@@ -106,14 +106,38 @@ namespace rocwmma
         struct Driver
         {
         private:
+            template <typename DataT,
+                      std::enable_if_t<sizeof(DataT) == sizeof(uint64_t), bool> = true>
+            ROCWMMA_DEVICE static inline auto dpp(DataT&& v0, DataT&& v1)
+            {
+                constexpr uint32_t B32VecSize = 2;
+                using B32VecT                 = VecT<uint32_t, B32VecSize>;
+
+                auto op = [](auto&& idx, auto&& v0, auto&& v1) {
+                    constexpr auto i = std::decay_t<decltype(idx)>::value;
+                    return DppOp::template exec<WriteRowMask, WriteBankMask, BoundCtrl>(v0.data[i],
+                                                                                        v1.data[i]);
+                };
+
+                auto dpp_result = vector_generator<uint32_t, B32VecSize>()(
+                    op, reinterpret_cast<B32VecT const&>(v0), reinterpret_cast<B32VecT const&>(v1));
+                return reinterpret_cast<DataT&>(dpp_result);
+            }
+
+            template <typename DataT,
+                      std::enable_if_t<sizeof(DataT) < sizeof(uint64_t), bool> = false>
```

Comment:
Might change comparison to !=, to make the two functions mutually exclusive. The internal b32 check on Dpp::exec will catch cases that are not supported

Reply:
> Might change comparison to !=, to make the two functions mutually exclusive. The internal b32 check on Dpp::exec will catch cases that are not supported

done. :)